nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -0,0 +1,535 @@
+    cmd_vel_out_topic_, 1);
+
+  if (polygon_topic_.length()) {
+    rclcpp::QoS polygon_qos = rclcpp::SystemDefaultsQoS();  // set to default
+    polygon_pub_ = this->create_publisher<geometry_msgs::msg::PolygonStamped>(
+      polygon_topic_, polygon_qos);
+  }
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+CollisionMonitorNode::on_activate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Activating");
+
+  // Activating lifecycle publishers
+  polygon_pub_->on_activate();
+  cmd_vel_out_pub_->on_activate();
+
+  // Activating main worker
+  setWorkerActive(true);
+
+  // Creating bond connection
+  createBond();
+
+  // Polygons publishing thread
+  polygon_pub_timer_ = this->create_wall_timer(
+    std::chrono::milliseconds(200),
+    std::bind(&CollisionMonitorNode::publishPolygons, this));
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+CollisionMonitorNode::on_deactivate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Deactivating");
+
+  // Deactivating main worker
+  setWorkerActive(false);
+
+  // Reset opeartion states to their defaults after worker deactivating
+  release_operation_ = false;
+  ra_prev_ = {DO_NOTHING, {0.0, 0.0, 0.0}};
+
+  // Deactivating lifecycle publishers
+  polygon_pub_->on_deactivate();
+  cmd_vel_out_pub_->on_deactivate();
+
+  // Deactivating polygons publishing timer
+  polygon_pub_timer_->cancel();
+
+  // Destroying bond connection
+  destroyBond();
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+CollisionMonitorNode::on_cleanup(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Cleaning up");
+
+  tf_listener_.reset();
+  tf_buffer_.reset();
+
+  odom_sub_.reset();
+  cmd_vel_in_sub_.reset();
+
+  polygon_pub_.reset();
+  cmd_vel_out_pub_.reset();
+
+  polygon_pub_timer_.reset();
+
+  polygons_.clear();
+  sources_.clear();
+
+  resetVelocity();
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+CollisionMonitorNode::on_shutdown(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Shutting down");
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+void CollisionMonitorNode::publishPolygons()
+{
+  if (!polygon_topic_.length()) {
+    // Topic is not specified: there is no need to publish
+    return;
+  }
+
+  // Fill PolygonStamped struct
+  std::unique_ptr<geometry_msgs::msg::PolygonStamped> poly_s =
+    std::make_unique<geometry_msgs::msg::PolygonStamped>();
+  poly_s->header.stamp = this->now();
+  poly_s->header.frame_id = base_frame_id_;
+  poly_s->polygon = polygons_pub_;
+
+  // Publish polygon
+  polygon_pub_->publish(std::move(poly_s));
+}
+
+void CollisionMonitorNode::odomCallback(nav_msgs::msg::Odometry::ConstSharedPtr msg)
+{
+  std::lock_guard<mutex_t> lock(velocity_mutex_);
+
+  double dt;
+  try {
```

Comment:
Remove try/catch, same with the source base with using `Duration`'s to make this all 1 line in L196

Reply:
[Done](https://github.com/ros-planning/navigation2/pull/2982/files#diff-e1c76d88179c14a1cf399e94bebd4f5d17dedca39e09b3bbdc810aa11a705543R186-R195)